src/depot.cpp

```diff
@@ -100,13 +100,16 @@ void Depot::AfterAddRemove(TileArea ta, bool adding)
 		this->ta.Add(tile);
 	}
 
+	VehicleType veh_type = this->veh_type;
 	if (this->ta.tile != INVALID_TILE) {
 		this->RescanDepotTiles();
 		assert(!this->depot_tiles.empty());
 		this->xy = this->depot_tiles[0];
 	} else {
 		delete this;
 	}
+
+	InvalidateWindowData(WC_SELECT_DEPOT, veh_type);
```

Comment:
You don't seem to use this variable anywhere else, so might as well save a line.

```suggestion
	InvalidateWindowData(WC_SELECT_DEPOT, this->veh_type);
```

Reply:
Probably used in the extended depots branch.
Nice spot.